moto/route53domains/responses.py

```diff
@@ -0,0 +1,143 @@
+import json
+from typing import Dict, List, Optional
+
+from moto.core.responses import BaseResponse
+from moto.route53domains.models import Route53DomainsBackend, route53domains_backends
+from moto.route53domains.validators import Route53Domain, Route53DomainsOperation
+
+
+class Route53DomainsResponse(BaseResponse):
+    def __init__(self) -> None:
+        super().__init__(service_name="route53-domains")
+
+    @property
+    def route53domains_backend(self) -> Route53DomainsBackend:
+        """Return backend instance"""
+        return route53domains_backends[self.current_account]["global"]
+
+    def register_domain(self) -> str:
+        """Register a domain"""
+        domain_name: Optional[str] = self._get_param("DomainName")
```

Comment:
```suggestion
        domain_name = self._get_param("DomainName")
```
The `_get_param`-operations returns `Any`, if I'm not mistaken. So if the only goal is to make `mypy` happy you could remove all the explicit types.

Reply:
Honestly it was really just for documentation's sake, so that anyone who reads the code in the future would know that type they should be. Anyway, I removed these type hints since they upset mypy as well